python/ray/train/trainer.py

```diff
@@ -566,14 +567,15 @@ class TrainingIterator:
     """An iterator over Train results. Returned by ``trainer.run_iterator``."""
 
     def __init__(
-            self, backend_executor_actor: ActorHandle,
+            self, backend_executor_actor: ActorHandle, backend: Backend,
             train_func: Union[Callable[[], T], Callable[[Dict[str, Any]], T]],
             run_dir: Path,
             dataset: Optional[Union[RayDataset, Dict[str, RayDataset]]],
             checkpoint_manager: CheckpointManager,
             checkpoint: Optional[Union[Dict, str, Path]],
             checkpoint_strategy: Optional[CheckpointStrategy]):
         self._backend_executor_actor = backend_executor_actor
+        self._backend = backend
```

Comment:
Is this being used anywhere?

Reply:
Ah, use this to pass in `decode_checkpoint` to `_process_checkpoint`?

Reply:
Fixed!